recipes/gettext/all/conanfile.py

```diff
@@ -0,0 +1,127 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, VisualStudioBuildEnvironment, tools
+import os
+import shutil
+import glob
+
+
+class GetTextConan(ConanFile):
+    name = "gettext"
+    description = "An internationalization and localization system for multilingual programs"
+    topics = ("conan", "gettext", "intl", "libintl", "i18n")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gnu.org/software/gettext"
+    license = "GPL-3.0-or-later"
+    settings = "os_build", "arch_build", "arch", "compiler", "build_type"
+    exports_sources = ["patches/*.patch"]
+
+    
+
+    requires = ("libiconv/1.15",
```

Comment:
Should this be a `require` or a `build_requires`?
Because users of this recipe will now have to eat this dependency as well.

Reply:
Also, these MUST be static, because when crossbuilding `arch` might not be equal to `arch_build`.

Reply:
I transformed these requirements into private requirements. It seems to me that it is the exact use case for this: libraries are completely embedded in the executables, and it does not matter if another version (or same version with different options) is present in the dependency graph.

Reply:
Good idea.
In reality, they should only be private if they are static libraries.
But the private flag of a dependency can (currently) not be changed afterwards? @uilianries

Reply:
as far as I understand, the"private" part makes the recipe create a new requirement node in the graph(using default options I guess ?), even if the requirement is already present somewhere in the graph.